internal/db/p2p_schema_root.go

```diff
@@ -205,3 +176,52 @@ func (p *Peer) GetAllP2PCollections(ctx context.Context) ([]string, error) {
 
 	return collectionIDs, nil
 }
+
+func (db *db) PeerInfo() peer.AddrInfo {
+	db.peerMutex.RLock()
+	defer db.peerMutex.RUnlock()
```

Comment:
todo: Why are you locking here?  Given that you are returning `db.peerInfo` outside of the `RUnlock` call, the lock appears pointless and creates a false sense of thread-safety.

Please either document why you are locking here, or fix the code so that it is obviously correct/expected (and probably documented on the `PeerInfo()` func signature)

Reply:
Change to an `atomic.Value`